app/controllers/badges_controller.rb

```diff
@@ -1,6 +1,7 @@
 class BadgesController < ApplicationController
   before_action :authenticate_user!
   before_action :set_badge, only: [:show]
+  load_and_authorize_resource except: [:require, :redeem]
```

Comment:
How about adding an `admin` boolean to the use and doing `can :manage, Badge if user.admin?`